src/topojsonm/topojsonm.ml

```diff
@@ -171,27 +171,33 @@ let map_object f src dst =
   in
   let rec go () =
     match Jsonm.decode decoder with
-    | `Lexeme (`Name "objects" as t) ->
-        (match Jsonm.decode decoder with
-        | `Os -> (
-        let b = Jsonm.decode decoder in
-            match Topo.of_json @@ decode_single_object decoder with
-            | Error (`Msg m) -> raise (Abort (`Unexpected m))
-            | Ok v -> (
-                match Topo.topojson v with
-                | Geometry g ->
-                    let g' = f b g in
-                    enc (`Lexeme t);
-                    encode_value encoder
-                      (Topo.to_json @@ Topo.v ?bbox:(Topo.bbox v) (Geometry g'));
-                    go ()
-                | _ -> raise (Invalid_argument "Expected a geometry object")))
+    | `Lexeme (`Name "objects" as t) -> (
+        match Jsonm.decode decoder with
+        | `Lexeme `Os ->
+            let rec loop_through_objects decoder =
```

Comment:
I wouldn't define it inside here but move it completely outside of this function.